modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/SleepTimerRestartWhenShakingDevice.kt

```diff
@@ -56,23 +62,37 @@ class SleepTimerRestartWhenShakingDevice @Inject constructor(
     }
 
     private fun onDeviceShaken() {
-        sleepTimer.restartTimerIfIsRunning onSuccess@{
+        val time = sleepTimer.restartTimerIfIsRunning onSuccess@{
             playbackManager.updateSleepTimerStatus(running = true, sleepAfterEpisode = false)
 
-            analyticsTracker.track(AnalyticsEvent.PLAYER_SLEEP_TIMER_RESTARTED, mapOf(TIME_KEY to SHAKING_PHONE_VALUE))
-
             if (context.isAppForeground()) {
                 Toast.makeText(
                     context,
                     context.getString(R.string.player_sleep_timer_restarted_after_device_shake),
                     Toast.LENGTH_SHORT,
                 ).show()
+            } else {
+                playbackManager.playSleepTimeTone()
             }
         }
+        time?.let { trackSleepTimeRestart(it) }
+    }
+
+    private fun trackSleepTimeRestart(time: Duration) {
+        val currentTime = System.currentTimeMillis().milliseconds
+        val elapsedTime: Duration? = lastTrackTime?.let { currentTime - it }
+        if (elapsedTime == null || elapsedTime >= 3.seconds) { // Make sure we don't send the same report in 3 seconds
+            analyticsTracker.track(
+                PLAYER_SLEEP_TIMER_RESTARTED,
+                mapOf(TIME_KEY to time.inWholeSeconds, REASON_KEY to DEVICE_SHAKE_VALUE),
+            )
+            lastTrackTime = currentTime
+        }
```

Comment:
I implemented this because I noticed we were reporting 10ish times per restart because of the device shaken detection